controllers/dataprotection/backuprepo_controller.go

```diff
@@ -414,6 +475,46 @@ func (r *BackupRepoReconciler) createStorageClass(
 		})
 }
 
+func (r *BackupRepoReconciler) checkPVCTemplate(reqCtx intctrlutil.RequestCtx,
+	renderCtx renderContext, repo *dpv1alpha1.BackupRepo, provider *storagev1alpha1.StorageProvider) error {
+
+	condType := ConditionTypePVCTemplateChecked
+	checkedTemplateMd5 := repo.Annotations[dataProtectionPVCTemplateMD5MD5AnnotationKey]
+	currentTemplateMd5 := md5Digest(provider.Spec.PersistentVolumeClaimTemplate)
+	if provider.Spec.PersistentVolumeClaimTemplate != "" && checkedTemplateMd5 != currentTemplateMd5 {
+		pvc := &corev1.PersistentVolumeClaim{}
+		err := r.constructPVCByTemplate(renderCtx, pvc, repo, provider.Spec.PersistentVolumeClaimTemplate)
+		if err != nil {
+			_ = updateCondition(reqCtx.Ctx, r.Client, repo, condType,
+				metav1.ConditionFalse, ReasonBadPVCTemplate, err.Error())
+			return err
+		}
+	}
+	if err := updateCondition(reqCtx.Ctx, r.Client, repo, condType,
+		metav1.ConditionTrue, ReasonPVCTemplateChecked, ""); err != nil {
+		return err
+	}
+	return updateAnnotations(reqCtx.Ctx, r.Client, repo, map[string]string{
+		dataProtectionPVCTemplateMD5MD5AnnotationKey: currentTemplateMd5,
+	})
+}
+
+func (r *BackupRepoReconciler) constructPVCByTemplate(
+	renderCtx renderContext, pvc *corev1.PersistentVolumeClaim,
+	repo *dpv1alpha1.BackupRepo, tmpl string) error {
+	// fill render values
+	renderCtx.GeneratedStorageClassName = repo.Status.GeneratedStorageClassName
```

Comment:
is it necessary for pvcTemplate?

Reply:
This allows a CSI based provider to specify a pvcTemplate to override some default settings, in this case it needs the name of the generated storage class. For example:

```yaml
apiVersion: storage.kubeblocks.io/v1alpha1
kind: StorageProvider
metadata:
  name: oss
spec:
  csiDriverName: ru.yandex.s3.csi
  csiDriverSecretTemplate: |
    accessKeyID: {{ index .Parameters "accessKeyId" }}
    secretAccessKey: {{ index .Parameters "secretAccessKey" }}
    ...

  storageClassTemplate: |
    provisioner: ru.yandex.s3.csi
    parameters:
      mounter: geesefs
      bucket: {{ index .Parameters "bucket" }}
      csi.storage.k8s.io/provisioner-secret-name: {{ .CSIDriverSecretRef.Name }}
      csi.storage.k8s.io/provisioner-secret-namespace: {{ .CSIDriverSecretRef.Namespace }}
      ...

  persistentVolumeClaimTemplate: |
    spec:
      storageClassName: {{ .GeneratedStorageClassName }}
      accessModes:
        # use ReadWriteOnce instead of the default one, which is ReadWriteMany
        - ReadWriteOnce
```